src/main/java/seedu/address/model/VersionedContentList.java

```diff
@@ -0,0 +1,103 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Stores previous Content versions since application initialisation
+ */
+public class VersionedContentList {
+    private final List<Content> contentVersions;
+
+    /**
+     * Constructor for the VersionedContactList object.
+     * @param initialContent initial content state of the app
+     */
+    public VersionedContentList(Content initialContent) {
+        this.contentVersions = new LinkedList<>();
```

Comment:
Any particular reason why a LinkedList is used instead of lets say, an ArrayList?

Reply:
I used a linked list as this data structure is expected to support a lot of adding and deleting of new content states. Linked lists are expected to perform better for such operations compared to arraylists